### PR TITLE
Webpack and typescript guide

### PR DESCRIPTION
**Pull request by morsdyce**
This PR relates to issue #196 
This PR is based upon difficulties I experienced setting up a project with webpack 2 and typescript (mainly the non code imports).

Please note that english is not my native language so this may include grammar and various mistakes.

### CODE REVIEW

File: content/guides/webpack-and-typescript.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
+---
+title: Webpack & Typescript
+sort: 20
+contributors:
+  - morsdyce
+---
+
+Typescript is a typed superset of javascript that compiles to plain javascript, in this guide we will learn how to integrate Typescript with webpack.
```
Comment (bebraw) [inline content/guides/webpack-and-typescript.md]: * /s/Typescript/TypeScript
* /s/javascript/JavaScript

File: content/guides/webpack-and-typescript.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+To learn more about Typescript [click here](https://www.typescriptlang.org).
```
Comment (bebraw) [inline content/guides/webpack-and-typescript.md]: You can replace this sentence by linking to TypeScript in the previous paragraph.

File: content/guides/webpack-and-typescript.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+## Basic Setup
+
+In order to get started with Webpack and Typescript, first we must install webpack in our project.
+If you didn't do so already please check out [Webpack Installation Guide](https://webpack.js.org/guides/installation/).
```
Comment (bebraw) [inline content/guides/webpack-and-typescript.md]: /s/Webpack/webpack

File: content/guides/webpack-and-typescript.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+To start using webpack with Typescript you need a couple of things:
+1. Install the Typescript compiler in your project.
+2. Install a Typescript loader. (in this case we're using ts-loader)
+3. Create a tsconfig.json file to contain our typescript compilation configuration.
```
Comment (bebraw) [inline content/guides/webpack-and-typescript.md]: Add formatting to filenames. See how other articles do it.

File: content/guides/webpack-and-typescript.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+to install the typing definition for that library.
+
+You can install 3rd party library definitions from the @types repository.
+
+For example if we want to install lodash we can run the following command to get the typings for it:
+`npm install --save-dev @types/lodash`
+
+For more information see [this blog post](https://blogs.msdn.microsoft.com/typescript/2016/06/15/the-future-of-declaration-files/)
+
+## Importing non code assets
```
Comment (bebraw) [inline content/guides/webpack-and-typescript.md]: This section probably belongs to another article as it goes beyond the original topic. Another PR would work if the content doesn't exist yet.

Comment (morsdyce) [inline content/guides/webpack-and-typescript.md]: I believe that at least the typescript related content in this section should stay in this page, it was a big issue for me learning to use webpack with typescript and there wasn't allot of resources on this specific topic.

Would you agree if I move the part that explains how to load non code assets which means the text until "Here we instruct webpack to include this file in our build and return a url reference in our Typescript code."

And then keep the rest of the content in this page, so people know how to configure typescript to work with custom file loaders?

Also I'm not sure where should the importing non code assets section can go to as I don't see anything existing right now, should it be it's own page?

Comment (bebraw) [inline content/guides/webpack-and-typescript.md]: > Also I'm not sure where should the importing non code assets section can go to as I don't see anything existing right now, should it be it's own page?

Yup. Most likely a guide.

You can show the TypeScript specific portion here while pushing the asset specific concerns elsewhere.

**Comment by bebraw** [comment]
Merged. Feel free to PR more specific tweaks etc. Thanks. 👍

**Pull request merged**